azure-devops/azext_devops/dev/pipelines/pipeline_create_helpers/pipelines_resource_provider.py

```diff
@@ -19,36 +19,36 @@ def get_github_service_endpoint(organization, project):
     This will try to create a GitHub service connection if there is no existing one in the project
     GitHub pat token will be asked for interactively or can be provided
     by setting the Environment variable AZ_DEVOPS_GITHUB_PAT_ENVKEY.
-    Service endpoint connection name is asked as input from the user, if the environment is non interative
-    name is set to default  AzureDevopsCliCreatePipelineFlow
+    Service endpoint connection name is asked as input from the user
     """
+    from azext_devops.dev.common.identities import get_current_identity, get_account_from_identity
+    authenticated_user_unique_id = get_account_from_identity(get_current_identity(organization))
     se_client = get_service_endpoint_client(organization)
     existing_service_endpoints = _get_service_endpoints(organization, project, 'github')
     service_endpoints_choice_list = ['Create new GitHub service connection']
     github_service_endpoints = []
     choice = 0
     for endpoint in existing_service_endpoints:
-        if endpoint.authorization.scheme == 'InstallationToken':
-            service_endpoints_choice_list.append('{} {}'.format(endpoint.name, '(Recommended)'))
-        else:
+        if (endpoint.authorization.scheme == 'InstallationToken' or
```

Comment:
`get_service_endpoints` in the client takes scheme as a param for filtering... can we use that one?

Reply:
I think I cannot use that filter here.. 

**Reason:** I need all service connections created by me or the ones with auth scheme = installationtoken. 
I will still need to pass all auth schemes to get_service_endpoints since I need my service endpoints which may or may not be InstallationToken scheme. 

Good catch on the functionality being available though. I had missed that.